data/transactions/payment.go

```diff
@@ -84,11 +84,16 @@ func (payment PaymentTxnFields) apply(header Header, balances Balances, spec Spe
 		}
 
 		// Confirm that we have no balance left
-		rec, err = balances.Get(header.Sender)
+		rec, err = balances.Get(header.Sender, true)
 		if !rec.AccountData.MicroAlgos.IsZero() {
 			return fmt.Errorf("balance %d still not zero after CloseRemainderTo", rec.AccountData.MicroAlgos.Raw)
 		}
 
+		// Confirm that there is no asset-related state in the account
+		if len(rec.Assets) > 0 {
```

Comment:
This appears to be sufficient since I don't think there's a way to make `len(rec.AssetParams) > 0` while `len(rec.Assets) == 0`, but it might be worth checking that `rec.AssetParams` is empty as well.

Reply:
Yeah, that was my thinking too (should be impossible), but happy to add a check just in case.